### ISSUE TITLE
Eliminate all hardcoded git dependencies

### ISSUE THREAD

**Issue by hdevalence** [ZcashFoundation/parity-zcash#72]
Right now we have a lot of hardcoded git dependencies on various repositories, left over from the Parity development.  It would be good to eliminate all of the git dependencies and replace them by crates-io published versions (which are immutable and so can't be modified out from under us).

**Comment by dconnolly** [comment]
>  immutable and so can't be modified out from under us

We could pin to a commit in the git dependencies for the time being.

**Comment by hdevalence** [comment]
I think if we're going to do the work of adding pins we might as well add released versions... adding a pin doesn for instance the reason that the build is broken in #70  is that someone merged code into https://github.com/ethcore/rust-secp256k1 and there's no versioning on our dependency, so now all the trait versions are out of sync and broken :(

**Comment by dconnolly** [comment]
Well for the git deps that aren't versioned that we don't control, we would have to fork them, right? Currently Dependabot notifies on new version availability and runs through the lock file to make the updates and then test the build, but it hasn't done all of them yet so as not to spam us.

**Comment by hdevalence** [comment]
No, I think that many of the dependencies that are there don't need to be git dependencies at all -- like the secp bindings -- or could be eliminated from our source tree.

**Comment by dconnolly** [comment]
Ha, at least `ethcore/rust-secp256k1` aka `paritytech/rust-secp256k1` is a fork of a fork of https://crates.io/crates/secp256k1 aka `rust-bitcoin/rust-secp256k1` 🙃🍴
